Mirror/Runtime/NetworkAnimator.cs

```diff
@@ -231,65 +231,63 @@ internal void HandleAnimTriggerMsg(int hash)
 
         void WriteParameters(NetworkWriter writer, bool autoSend)
         {
-            for (int i = 0; i < m_Animator.parameters.Length; i++)
+            AnimatorControllerParameter[] parameters = m_Animator.parameters;
+
+            for (int i = 0; i < parameters.Length; i++)
             {
                 if (autoSend && !GetParameterAutoSend(i))
                     continue;
 
-                AnimatorControllerParameter par = m_Animator.parameters[i];
-                if (par.type == AnimatorControllerParameterType.Int)
-                {
-                    writer.WritePackedUInt32((uint)m_Animator.GetInteger(par.nameHash));
-
-                    SetSendTrackingParam(par.name + ":" + m_Animator.GetInteger(par.nameHash), i);
-                }
-
-                if (par.type == AnimatorControllerParameterType.Float)
+                AnimatorControllerParameter par = parameters[i];
+                switch (par.type)
                 {
-                    writer.Write(m_Animator.GetFloat(par.nameHash));
-
-                    SetSendTrackingParam(par.name + ":" + m_Animator.GetFloat(par.nameHash), i);
-                }
-
-                if (par.type == AnimatorControllerParameterType.Bool)
-                {
-                    writer.Write(m_Animator.GetBool(par.nameHash));
-
-                    SetSendTrackingParam(par.name + ":" + m_Animator.GetBool(par.nameHash), i);
+                    case AnimatorControllerParameterType.Int:
```

Comment:
can you please submit without the switch again so it's only the parameter caching?
no magic :)

Reply:
In my opinion the switch statement is better here because it more clearly shows the intention of the program. A series of if statements suggests that more than one case can be true. If you dislike the switch statement, I suggest we at least use `if-else` instead.